src/modules/compliance/src/lib/procedures/ensureFilePermissions.cpp

```diff
@@ -16,41 +16,56 @@ namespace compliance
 AUDIT_FN(ensureFilePermissions, "filename:Path to the file:M", "owner:Required owner of the file", "group:Required group of the file",
     "permissions:Required octal permissions of the file::^[0-7]{3,4}$", "mask:Required octal permissions of the file - mask::^[0-7]{3,4}$")
 {
-    UNUSED(log);
     struct stat statbuf;
     if (args.find("filename") == args.end())
     {
-        return Error("No filename provided");
+        OsConfigLogError(log, "No filename provided");
+        return Error("No filename provided", EINVAL);
     }
-    logstream << "ensureFilePermissions for '" << args["filename"] << "' ";
+    const auto filename = args["filename"];
```

Comment:
Style thing - since we're doing this thing everywhere, maybe we can agree on a single style, e.g. I like:
```
auto it = args.find("foo")
if (it == args.end()) {
   (errro)
}
filename = std::move(it->second)

Or, for non-mandatory

if (it != end)
{ 
 filename = std::move...
 
 }